app/src/main/java/com/github/multimatum_team/multimatum/model/datetime_parser/DateTimePatterns.kt

```diff
@@ -0,0 +1,215 @@
+            Token::asHour24,
+            Token::asTimeSeparator,
+            Token::asMinute
+        ) to { args: List<Any?> ->
+            timeFor(hour = args.getInt(0), minute = args.getInt(2))
+        }
+
+    private val `3am` =
+        listOf(
+            Token::asHour12,
+            tokenMatchingOneOf("am")
+        ) to { args: List<Any?> ->
+            timeFor(hour = args.getInt(0), minute = 0)
+        }
+
+    private val `3pm` =
+        listOf(
+            Token::asHour12,
+            tokenMatchingOneOf("pm")
+        ) to { args: List<Any?> ->
+            timeFor(hour = args.getInt(0) + 12, minute = 0)
+        }
+
+    private val `1-01-2000` =
+        listOf(
+            Token::asPossibleDayOfMonthIndex,
+            Token::asDateSeparator,
+            Token::asMonth,
+            Token::asDateSeparator,
+            Token::asPossibleYear
+        ) to { args: List<Any?> ->
+            dateForIfExists(year = args.getInt(4), month = args.getMonth(2), day = args.getInt(0))
+        }
+
+    private val `2000-1-1` =
+        listOf(
+            Token::asPossibleYear,
+            Token::asDateSeparator,
+            Token::asMonth,
+            Token::asDateSeparator,
+            Token::asPossibleDayOfMonthIndex
+        ) to { args: List<Any?> ->
+            dateForIfExists(day = args.getInt(4), month = args.getMonth(2), year = args.getInt(0))
+        }
+
+    private val `1_01_2000` =
+        listOf(
+            Token::asPossibleDayOfMonthIndex,
+            Token::asMonth,
+            Token::asPossibleYear
+        ) to { args: List<Any?> ->
+            dateForIfExists(day = args.getInt(0), month = args.getMonth(1), year = args.getInt(2))
+        }
+
+    private val monday =
+        listOf(
+            Token::asDayOfWeek
+        ) to { args: List<Any?> ->
+            val requestedDayOfWeek = args.getDayOfWeek(0)
+            dateForNextDayMatching(requestedDayOfWeek)
+        }
+
+    private val monday_15 =
+        listOf(
+            Token::asDayOfWeek,
+            Token::asPossibleDayOfMonthIndex
+        ) to { args: List<Any?> ->
+            val targetDayOfWeek = args.getDayOfWeek(0)
+            val dayOfMonthIdx = args.getInt(1)
+            val currentDate = currentDateProvider()
+            val targetDate =
+                localDateForIfExists(currentDate.year, currentDate.month, dayOfMonthIdx)
+            targetDate?.let {
+                if (targetDate.dayOfWeek == targetDayOfWeek) ExtractedDate(targetDate)
+                else null
+            }
+        }
+
+    private val midday =
+        listOf(
+            tokenMatchingOneOf("noon", "midday")
+        ) to { _: List<Any?> ->
+            timeFor(12, 0)
+        }
+
+    private val midnight =
+        listOf(
+            tokenMatchingOneOf("midnight")
+        ) to { _: List<Any?> ->
+            timeFor(0, 0)
+        }
+
+    val patterns: List<Pair<List<(Token) -> Any?>, (List<Any?>) -> ExtractedInfo?>> =
```

Comment:
Great job ! However, I have a question:
Do you think there could be a problem if we write a date with the english pattern (the month before the day) ?

Reply:
It depends on what you write exactly... E.g. `2022-1-1` would be parsed as the 1st of January 2022, but some other patterns would very likely be missed. Anyway I will probably add some additional patterns next week, so actually if you have some time to try patterns that I might have missed and give me a list of the ones you think I should add, that would be really helpful!